src/ompl/util/RandomNumbers.h

```diff
@@ -139,6 +146,30 @@ namespace ompl
             return localSeed_;
         }
 
+        /** \brief Uniform random sampling of a unit-length vector. I.e., the surface of an n-ball */
+        void uniformNormalVector(unsigned int n, double value[]);
+
+        /** \brief Uniform random sampling of the content of an n-ball, with a radius appropriately distributed between [0,r) so that the distribution is uniform in a Cartesian coordinate system. */
+        void uniformInBall(double r, unsigned int n, double value[]);
+
+        /** \brief Uniform random sampling of the surface of a prolate hyperspheroid, a special symmetric type of
+        n-dimensional ellipse.
+        @par J D. Gammell, S. S. Srinivasa, T. D. Barfoot, "Informed RRT*: Optimal Sampling-based
+        Path Planning Focused via Direct Sampling of an Admissible Ellipsoidal Heuristic."
+        IROS 2014. DOI: <a href="http://dx.doi.org/10.1109/IROS.2014.6942976">10.1109/IROS.2014.6942976</a>.
+        <a href="http://www.youtube.com/watch?v=d7dX5MvDYTc">Illustration video</a>.
+        <a href="http://www.youtube.com/watch?v=nsl-5MZfwu4">Short description video</a>. */
+        void uniformProlateHyperspheroidSurface(ProlateHyperspheroidPtr phsPtr, unsigned int n, double value[]);
```

Comment:
I didn't want to create a default dimension to the PHS class, so there is no default constructor. As a result I figured most classes would create a `shared_ptr` to the class instead of owning the class itself... Which is why I wrote the function to take a `shared_ptr`.

Given that, is there still a benefit to making the function take the class and just having using functions, i.e., `PathLengthDirectInfSampler`, dereference their pointer? I guess it may reduce the number of dereference calls inside the `RNG` function?